### DISCUSSION TITLE
Flashlist and Pagination

### DISCUSSION THREAD

**Discussion by XdieselX** [Shopify/flash-list#722] [Q&A]
Hi,

I would like to know how to implement infinite scroll without loading everything. I have been using onEndReached with FlatList the distancefromend argument to calculate the distance left before the next page is loaded.

Now, the Flashlist is loading every page one by one even though onEndReachedThreshold={0.5}.

Thank you in advance for your help.

New Adopter.

**Comment by parthpvaghani** [comment]
I've also faced this same problem

**Comment by Bayramito** [comment]
In my case i use a helper library to handle my requests claled mobx-observable-promise... 

every promise in it has a prop called `hasMore`.  
What it does is, initially it is sending 1 additional request when your scrolling is done, to check if there is more items to come...
if so, on your next `onEndReached` call it sends the request and repeat this process again...

if not, then it's not calling anything... but your data source should have a valid pagination ofc... to be able to work with this...

NOTE: make sure your fflashlist in a `flex:1` View or something so it can take entire blank space to make its initial calculations properly
